src/main/java/com/relayrides/pushy/apns/ApnsClient.java

```diff
@@ -443,6 +449,45 @@ protected void handshakeFailure(final ChannelHandlerContext context, final Throw
     }
 
     /**
+     * Returns socket channel class suitable for specified event loop group.
+     *
+     * @param eventLoopGroup
+     * @return socket channel class
+     * @throws IllegalArgumentException in case of null or unrecognized event loop group.
+     */
+    private Class<? extends Channel> getSocketChannelClass(EventLoopGroup eventLoopGroup) {
+        if (eventLoopGroup == null) {
+            log.warn("Asked for socket channel class to work with null event loop group, returning NioSocketChannel class.");
+            return NioSocketChannel.class;
+        }
+
+        if (eventLoopGroup instanceof NioEventLoopGroup) {
+            return NioSocketChannel.class;
+        } else if (eventLoopGroup instanceof OioEventLoopGroup) {
+            return OioSocketChannel.class;
+        }
+
+        // epoll?
+        String className = eventLoopGroup.getClass().getName();
+        if (EPOLL_EVENT_LOOP_GROUP_CLASS.equals(className)) {
```

Comment:
To make sure I understand: are we not doing `instanceof` here because the epoll classes live in a separate library, and may not be on the classpath? If that's the case, would it make sense to use the same approach (matching class names) the whole way through for consistency? We could have a map of event loop group class names to socket class names, or could use a `switch` block. At the very least, could we add a comment explaining why we need to handle epoll differently?